.github/workflows/ci-cd-pull-request.yml

```diff
@@ -6,9 +6,82 @@ env:
   BASE_IMAGE: ghcr.io/graphprotocol/graph-docs-staging
 
 jobs:
+  lint:
+    if: contains(github.head_ref, 'crowdin') == false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 20
+          packageManager: pnpm
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Lint
+        run: pnpm lint
+
+  format:
+    if: contains(github.head_ref, 'crowdin') == false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 20
+          packageManager: pnpm
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Check formatting
+        run: pnpm prettier:check
+
+  typecheck:
+    if: contains(github.head_ref, 'crowdin') == false
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up env
+        uses: the-guild-org/shared-config/setup@main
+        with:
+          nodeVersion: 20
+          packageManager: pnpm
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Typecheck
+        run: pnpm typecheck
+
   build:
     if: contains(github.head_ref, 'crowdin') == false
     runs-on: ubuntu-latest
+    needs: [lint, format, typecheck]
```

Comment:
Do we still need to do this step below now that we added these as dependencies?

```
      - name: Lint and typecheck
        run: pnpm check
```

Reply:
yeah not needed